lib/util/config_utils.js

```diff
@@ -127,4 +127,29 @@ shaka.util.ConfigUtils = class {
     last[fieldName.substring(nameStart).replace(/\\\./g, '.')] = value;
     return configObject;
   }
+
+  /**
+   * Reference the input parameters so the compiler doesn't remove them from
+   * the calling function.  Return whatever value is specified.
+   *
+   * This allows an empty or default implementation of a config callback that
+   * still bears the complete function signature even in compiled mode.
+   *
+   * The caller should look something like this:
+   *
+   *   const callback = (a, b, c, d) => {
+   *     return referenceParametersAndReturn(
+             [a, b, c, d],
+             a);  // Can be anything, doesn't need to be one of the parameters
+   *   };
+   *
+   * @param {!Array.<?>} parameters
+   * @param {T} returnValue
+   * @return {T}
+   * @template T
+   * @noinline
+   */
+  static referenceParametersAndReturn(parameters, returnValue) {
+    return parameters && returnValue;
```

Comment:
It doesn't come up in any situation where we actually call this at the moment, but technically this approach won't work correctly if `returnValue` is falsey.

Reply:
It definitely will.  `true && 0 === 0`, `true && undefined === undefined`, etc.  It works this way for any truthy value, not just `true`.  And `parameters` (being a non-nullable Array) is always truthy, even if it's an empty array for some reason.